pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -169,8 +174,91 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		#[pallet::call_index(0)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-		pub fn create_pool(_origin: OriginFor<T>) -> DispatchResult {
-			todo!()
+		pub fn create_pool(
+			origin: OriginFor<T>,
+			curve: CurveInput<CurveParameterInputOf<T>>,
+			currencies: BoundedVec<TokenMetaOf<T>, T::MaxCurrencies>,
+			state: PoolStatus<Locks>,
+			denomination: u8,
+			pool_manager: AccountIdOf<T>,
+			transferable: bool,
+			team: Team<AccountIdOf<T>>,
```

Comment:
Honestly, this feels like quite a big extrinsic. Have you guys considered splitting this up into multiple, atomic, extrinsics, that can be batched together to achieve the same result? Especially setting the team, for instance. Once the pool is created with a manager, the next extrinsic could be called. I think it makes it easier. We've done this mistake once in the DID pallet and it's been biting us back to this day.

Reply:
The `transferable` property I understand if it cannot be changed after creation, but for the team I would imagine there could be need to update its members. Same for the pool state. I would suggest to take out of this extrinsic anything that could be updated in a different extrinsic.

Reply:
yeah 100% agree and we had talked about this prior. there really is no reason to not set this as part of dedicated extrinsics

Reply:
concerns team and pool manager, for transferable it's probably better we keep if for now

Reply:
I moved the reset tx to its own extrinsic in: [0f8c286](https://github.com/KILTprotocol/kilt-node/pull/768/commits/0f8c286505b02e167123d6d89164437d34a17fe9)